File: UI/travle_core/lib/src/models/tour_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'tour_update_request.g.dart';

/// Edit of an existing tour (mirrors the backend `TourUpdateRequest`). Content
/// only — activation state is toggled through the deactivate/activate endpoints,
/// and the organizer is never reassigned. [destinationIds] is the full desired
/// itinerary (the list order is the new sort order).
@JsonSerializable()
class TourUpdateRequest {
  TourUpdateRequest({
    required this.name,
    required this.description,
    required this.durationMinutes,
    required this.pricePerPerson,
    required this.capacity,
    required this.tourTypeId,
    this.destinationIds = const [],
  });

  final String name;
  final String description;
  final int durationMinutes;
  final double pricePerPerson;
  final int capacity;
  final int tourTypeId;
  final List<int> destinationIds;

  factory TourUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$TourUpdateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$TourUpdateRequestToJson(this);
}

File: Backend/Travle.Model/Requests/TourUpdateRequest.cs
namespace Travle.Model.Requests
{
    /// <summary>
    /// Edit of an existing tour by its organizer (or an admin). Content only — activation state is
    /// toggled through the dedicated deactivate/activate endpoints, and the organizer is never
    /// reassigned. <see cref="DestinationIds"/> is the full desired itinerary (the list order is the
    /// new <c>SortOrder</c>); the service reconciles stops against it. Changing
    /// <see cref="DurationMinutes"/> affects only schedules created afterwards — existing slots keep the
    /// end time they were published with.
    /// </summary>
    public class TourUpdateRequest
    {
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        public int DurationMinutes { get; set; }
        public decimal PricePerPerson { get; set; }

        public int Capacity { get; set; }

        public int TourTypeId { get; set; }

        public List<int> DestinationIds { get; set; } = new List<int>();
    }
}
